File: test/contract/coordinator_contract_test.go
package contract

import (
	"testing"

	agentv1 "openforge/gen/go/agent/v1"
	"github.com/stretchr/testify/require"
)

func TestContract_CoordinatorCreateAgent_RequestShape(t *testing.T) {
	golden := loadGolden(t, "golden/coordinator_create_agent.req.json")
	req := &agentv1.CreateAgentRequest{}
	assertRoundTrip(t, golden, req)

	require.Equal(t, "agent-001", req.GetAgentId())
	require.Equal(t, "p-001", req.GetPipelineId())
	require.Equal(t, "proj-1", req.GetProjectId())
	require.Equal(t, agentv1.AgentRole_AGENT_ROLE_WORKER, req.GetRole())
	require.Equal(t, "dev", req.GetMetadata()["env"])
	require.Equal(t, "team-a", req.GetMetadata()["owner"])
}

func TestContract_CoordinatorCreateAgent_ResponseShape(t *testing.T) {
	golden := loadGolden(t, "golden/coordinator_create_agent.resp.json")
	resp := &agentv1.CreateAgentResponse{}
	assertRoundTrip(t, golden, resp)

	require.Equal(t, "agent-001", resp.GetAgentId())
	require.Equal(t, "RUNNING", resp.GetStatus())
}

File: nodejs-io/src/contract/coordinator_contract.test.ts
import { describe, it, expect } from "vitest";
import { create } from "@bufbuild/protobuf";
import {
  CreateAgentRequestSchema,
  CreateAgentResponseSchema,
  type CreateAgentRequest,
  type CreateAgentResponse,
} from "../gen/agent/v1/coordinator_pb";
import { loadGolden } from "./golden_loader.js";

describe("Coordinator Contract (CreateAgent)", () => {
  it("CreateAgentRequest maps golden fields onto camelCase logical fields", () => {
    const golden = loadGolden("coordinator_create_agent.req.json") as {
      agent_id: string;
      pipeline_id: string;
      project_id: string;
      role: string;
      metadata: Record<string, string>;
    };

    const req: CreateAgentRequest = create(CreateAgentRequestSchema, {
      agentId: golden.agent_id,
      pipelineId: golden.pipeline_id,
      projectId: golden.project_id,
      role: golden.role as CreateAgentRequest["role"],
      metadata: golden.metadata,
    });

    expect(req.agentId).toBe("agent-001");
    expect(req.pipelineId).toBe("p-001");
    expect(req.projectId).toBe("proj-1");
    expect(req.role).toBe("AGENT_ROLE_WORKER");
    expect(req.metadata.env).toBe("dev");
    expect(req.metadata.owner).toBe("team-a");
  });

  it("CreateAgentResponse maps golden fields onto camelCase logical fields", () => {
    const golden = loadGolden("coordinator_create_agent.resp.json") as {
      agent_id: string;
      status: string;
    };

    const resp: CreateAgentResponse = create(CreateAgentResponseSchema, {
      agentId: golden.agent_id,
      status: golden.status,
    });

    expect(resp.agentId).toBe("agent-001");
    expect(resp.status).toBe("RUNNING");
  });
});
